storage/store.go

```diff
@@ -148,20 +148,33 @@ func (e *rangeAlreadyExists) Error() string {
 	return fmt.Sprintf("range for Raft ID %d already exists on store", e.rng.Desc().RaftID)
 }
 
-// A RangeSlice is a slice of Range pointers used for replica lookups
-// by key.
-type RangeSlice []*Range
+// rangeKeyItem is a common interface for proto.Key and Range.
+type rangeKeyItem interface {
+	getKey() proto.Key
+}
+
+// rangeBTreeKey is a type alias of proto.Key that implements the
+// rangeKeyItem interface and the btree.Item interface.
+type rangeBTreeKey proto.Key
 
-// Implementation of sort.Interface which sorts by StartKey from each
-// range's descriptor.
-func (rs RangeSlice) Len() int {
-	return len(rs)
+func (k rangeBTreeKey) getKey() proto.Key {
+	return (proto.Key)(k)
 }
-func (rs RangeSlice) Swap(i, j int) {
-	rs[i], rs[j] = rs[j], rs[i]
+
+func (k rangeBTreeKey) Less(i btree.Item) bool {
+	return k.getKey().Less(i.(rangeKeyItem).getKey())
 }
-func (rs RangeSlice) Less(i, j int) bool {
-	return bytes.Compare(rs[i].Desc().StartKey, rs[j].Desc().StartKey) < 0
+
+// rangeBTreeItem is a type alias of Range that implements the rangeKeyItem
+// interface and the btree.Item interface.
+type rangeBTreeItem Range
+
+func (r *rangeBTreeItem) getKey() proto.Key {
+	return (*Range)(r).Desc().EndKey
```

Comment:
is there any particular reason to key the tree by `EndKey` rather than `StartKey`?